### PR TITLE
Add more considerations/ steps to Software upgrade guide

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/1125
Resolves https://github.com/astronomer/docs/issues/1123

I'll push this to other Software versions once we confirm the steps for upgrading Deployments for the Kubernetes considerations.

### CODE REVIEW

File: software/upgrade-astronomer.md [hunk 1]
```diff
@@ -139,13 +139,31 @@ If there is a problem creating a new Airflow Deployment, check the commander log
 
 Make changes as needed and rerun the upgrade command from Step 7. Do not continue to Step 8 until you have successfully created a new Airflow Deployment.
 
+## Step 9: Upgrade the Astro CLI
+
+Each Software version is compatible only with specific versions of the Astro CLI. Ensure that all users on your installation are using the latest compatible version of the Astro CLI for your Software version. See [Version compatibility reference](version-compatibility-reference.md).
+
+To upgrade from a pre-1.0 version of the CLI to version 1.0+, see [Upgrade to Astro CLI version 1.0+](upgrade-astro-cli.md).
```
Comment (jwitz) [inline software/upgrade-astronomer.md:146]:
```suggestion
For standard upgrade steps, see [Upgrade the CLI](install-cli.md#upgrade-the-cli). To upgrade from a pre-1.0 version of the CLI to version 1.0+, see [Upgrade to Astro CLI version 1.0+](upgrade-astro-cli.md).
```

File: software/upgrade-astronomer.md [hunk 2]
```diff
@@ -139,13 +139,31 @@ If there is a problem creating a new Airflow Deployment, check the commander log
...
 
 To avoid extended service disruptions, Astronomer recommends upgrading Astronomer Software to a compatible version before you upgrade Kubernetes. To view Astronomer Software and Kubernetes compatibility information, see [Version compatibility reference for Astronomer Software](version-compatibility-reference.md#astronomer-software).
 
-### Upgrading to 0.30
+### Upgrading to Kubernetes 1.22
+
+If upgrade to Astronomer Software 0.29+ at the same time as you're upgrading to Kubernetes 1.22, complete your upgrades in the following order:
+
+1. Follow the standard Software upgrade procedure as described in this document.
+2. Upgrade all of your Deployments to use at least version 1.6.0 of the Airflow Helm chart. To do this, run the following command:
```
Comment (jwitz) [inline software/upgrade-astronomer.md:159]: @rbankston can you provide specific steps for how to do step 2 here? Should we also suggest for them to remove `--set astronomer.houston.upgradeDeployments.enabled=false \` from their upgrade command?

Comment (rbankston) [reply software/upgrade-astronomer.md:159]: kubectl exec -it `kubectl get pods -l component=houston --no-headers -n <release namespace>` -n <release namespace> -- yarn run upgrade-deployments

File: software/install-cli.md [hunk 3]
```diff
@@ -108,3 +108,58 @@ If the installation was successful, you should see the following output:
...
+    defaultValue="mac"
+    groupId= "upgrade-the-cli"
+    values={[
+        {label: 'Mac', value: 'mac'},
+        {label: 'Windows', value: 'windows'},
+        {label: 'Linux', value: 'linux'},
+    ]}>
+<TabItem value="mac">
+
+To upgrade the Astro CLI to the latest version, you can run the same command as you did to install the CLI for the first time:
```
Comment (kwcanuck) [inline software/install-cli.md:125]:
```suggestion
To upgrade the Astro CLI to the latest version, run the following command:
```

File: software/install-cli.md [hunk 4]
```diff
@@ -108,3 +108,58 @@ If the installation was successful, you should see the following output:
...
+
+</TabItem>
+
+<TabItem value="windows">
+
+To upgrade the Astro CLI on Windows:
+
+1. Delete the existing `astro.exe` file on your machine.
+
+2. Go to the [**Releases** page of the Astro CLI GitHub repository](https://github.com/astronomer/astro-cli/releases). Based on the version of the CLI you want and your CPU architecture, download one of the `.zip` files available on this page.
```
Comment (kwcanuck) [inline software/install-cli.md:139]:
```suggestion
2. Go to the [Astro CLI Releases page](https://github.com/astronomer/astro-cli/releases) and then download an Astro CLI `.zip` file. Astronomer recommends downloading the latest version to take advantage of the latest features and functionality.
```
I don't think _CPU architecture_ is necessary or relevant here.

File: software/install-cli.md [hunk 5]
```diff
@@ -108,3 +108,58 @@ If the installation was successful, you should see the following output:
...
+
+     For example, to upgrade to v1.0.0 of the Astro CLI on a Windows machine with an AMD 64 architecture, you download `astro_1.0.0-converged_windows_amd64.zip`.
```
Comment (kwcanuck) [inline software/install-cli.md:141]:
```suggestion
     For example, to upgrade to v1.0.0 of the Astro CLI on a Windows machine with an AMD 64 architecture, you download `astro_1.0.0-converged_windows_amd64.zip`.
```
Again, this doesn't seem relevant. From what I can tell, the zip files are organized by version only.

File: software/install-cli.md [hunk 6]
```diff
@@ -108,3 +108,58 @@ If the installation was successful, you should see the following output:
...
+
+3. If the `.zip` file does not automatically unzip, run the following command to unzip the executable:
```
Comment (kwcanuck) [inline software/install-cli.md:143]:
```suggestion
3. If the `.zip` file isn't automatically extracted, run the following command to extract the executable:
```
image

File: software/upgrade-astronomer.md [hunk 7]
```diff
@@ -139,13 +139,31 @@ If there is a problem creating a new Airflow Deployment, check the commander log
 
 Make changes as needed and rerun the upgrade command from Step 7. Do not continue to Step 8 until you have successfully created a new Airflow Deployment.
 
+## Step 9: Upgrade the Astro CLI
+
+Each Software version is compatible only with specific versions of the Astro CLI. Ensure that all users on your installation are using the latest compatible version of the Astro CLI for your Software version. See [Version compatibility reference](version-compatibility-reference.md).
```
Comment (kwcanuck) [inline software/upgrade-astronomer.md:144]:
```suggestion
Each Software version is compatible only with specific versions of the Astro CLI. Ensure that all users in your organization are using the latest compatible version of the Astro CLI for your Software version. See [Version compatibility reference](version-compatibility-reference.md).
```

File: software/upgrade-astronomer.md [hunk 8]
```diff
@@ -139,13 +139,31 @@ If there is a problem creating a new Airflow Deployment, check the commander log
...
 ## Upgrade considerations
 
 This topic contains information about upgrading to specific versions of Astronomer Software. This includes notes on breaking changes, database migrations, and other considerations that might depend on your use case.
 
 To avoid extended service disruptions, Astronomer recommends upgrading Astronomer Software to a compatible version before you upgrade Kubernetes. To view Astronomer Software and Kubernetes compatibility information, see [Version compatibility reference for Astronomer Software](version-compatibility-reference.md#astronomer-software).
 
-### Upgrading to 0.30
+### Upgrading to Kubernetes 1.22
+
+If upgrade to Astronomer Software 0.29+ at the same time as you're upgrading to Kubernetes 1.22, complete your upgrades in the following order:
```
Comment (kwcanuck) [inline software/upgrade-astronomer.md:156]:
```suggestion
If you're upgrading to Astronomer Software 0.29 or later and Kubernetes 1.22 at the same time, complete your upgrades in the following order:
```

File: software/upgrade-astronomer.md [hunk 9]
```diff
@@ -139,13 +139,31 @@ If there is a problem creating a new Airflow Deployment, check the commander log
...
+2. For each Deployment, run the following command to upgrade the Deployment to use the latest version of the Airflow Helm chart:
+
+    ```sh
+    kubectl exec -it `kubectl get pods -l component=houston --no-headers -n <deployment-namespace>` -n <deployment-namespace> -- yarn run upgrade-deployments
+    ```
+3. Upgrade Kubernetes to version 1.22.
+
+### Upgrading to Astronomer Software 0.30
 
 #### Running the 0.30 upgrade script with --no-hook
```
Comment (kwcanuck) [inline software/upgrade-astronomer.md:168]:
```suggestion
#### Run the 0.30 upgrade script with --no-hook
```

File: software/upgrade-astronomer.md [hunk 10]
```diff
@@ -159,7 +177,7 @@ A change in 0.30 enabled the `trgm` extension for PosgreSQL. If you use Azure Da
 
 If you don't complete this setup before your upgrade, the upgrade will fail.
 
-### Upgrading to 0.29
+### Upgrading to Astronomer Software 0.29
```
Comment (kwcanuck) [inline software/upgrade-astronomer.md:180]:
```suggestion
### Upgrade to Astronomer Software 0.29
```

File: software/upgrade-astronomer.md [hunk 11]
```diff
@@ -188,7 +206,7 @@ kubectl -n <your-platform-namespace> annotate secret astronomer-houston-jwt-sign
 If you upgraded to Astronomer Software 0.29 without annotating this secret, you can still complete the sync by running the following command after the upgrade:
 
 ```bash
-kubectl create job -n <release-namespace> --from=cronjob/astronomer-config-syncer upgrade-config-synchronization
+kubectl create job -n <your-platform-namespace> --from=cronjob/astronomer-config-syncer upgrade-config-synchronization
 ```
 
 ### Upgrading to Astronomer Software 0.28
```
Comment (kwcanuck) [inline software/upgrade-astronomer.md:212]:
```suggestion
### Upgrade to Astronomer Software 0.28
```

File: software/upgrade-astronomer.md [hunk 12]
```diff
@@ -188,7 +206,7 @@ kubectl -n <your-platform-namespace> annotate secret astronomer-houston-jwt-sign
 If you upgraded to Astronomer Software 0.29 without annotating this secret, you can still complete the sync by running the following command after the upgrade:
```
Comment (kwcanuck) [inline software/upgrade-astronomer.md:206]:
```suggestion
If you upgraded to Astronomer Software 0.29 without annotating this secret, run the following command to complete the synchronization:
```

**Review by kwcanuck — APPROVED** [review]

File: software/install-cli.md (hunk 4 above)
Comment (jwitz) [inline software/install-cli.md:139]:
```suggestion
2. Go to the [**Releases** page of the Astro CLI GitHub repository](https://github.com/astronomer/astro-cli/releases). Based on the version of the CLI you want and your CPU architecture, download one of the `.zip` files available in the **Assets** menu.
```

Comment (jwitz) [reply software/install-cli.md:139]: That's not quite right, you need to choose a download with either `amd` or `arm` which is based on your CPU architecutre
Screen Shot 2022-10-28 at 12 02 45 PM

File: software/install-cli.md [hunk 13]
```diff
@@ -108,3 +108,58 @@ If the installation was successful, you should see the following output:
 
 <pre><code parentName="pre">{`% astro version
 Astro CLI Version: ${siteVariables.cliVersion}`}</code></pre>
+
```
Comment (jwitz) [inline software/install-cli.md:111]:
```suggestion
```

**Pull request merged**